src/components/GettingStartedModal/index.tsx

```diff
@@ -0,0 +1,180 @@
+  ReactElement,
+  ReactNode,
+  SetStateAction,
+  useState,
+} from "react";
+import { Button } from "../Button";
+import { Modal } from "../Modal";
+import { PhoneticsStep } from "./PhoneticsStep";
+import { GroupRegistrationStep } from "./GroupRegistrationStep";
+import { PhoneticsPreference } from "../../state/reducers/phoneticsPreference";
+import { ChooseSetStep } from "./ChooseSetStep";
+import { StepIndicator } from "./StepIndicator";
+import { Preamble } from "./Preamble";
+import { useUserStateContext } from "../../providers/UserStateProvider";
+import { UserInteractors } from "../../state/useUserState";
+
+export interface WizardState {
+  // gives us information about the state of the entire wizard
+  groupId: string;
+  email: string;
+  whereFound: string;
+  phoneticsPreference?: PhoneticsPreference;
+  collectionId?: string; 
+}
+
+export interface StepProps {
+  // these props let us keep track of the user's data
+  // partial means any field can be undefined
+  wizardState: Partial<WizardState>;
+  setWizardState: Dispatch<SetStateAction<Partial<WizardState>>>;
+
+  // these functions let us navigate through the steps
+  // undefiend if on first step
+  goToPreviousStep?: () => void;
+  goToNextStep?: () => void; 
+  exitWizard?: () => void;
+}
+
+export interface Step {
+  title: string;
+  Component: (props: StepProps) => ReactElement;
+  /**
+   * This function says how we should actually update user state.
+   */
+  commitState: (state: WizardState, interactors: UserInteractors) => void;
+}
+
+/**
+ * This is the list of steps for the getting started modal.
+ *
+ */
+const steps: Step[] = [Preamble, GroupRegistrationStep, PhoneticsStep, ChooseSetStep];
+
+export function GettingStartedModal() {
+  const userStateContext = useUserStateContext();
+
+  // keep track of which step of workflow we are on (start on first step)
+  const [stepNumber, setStepNumber] = useState(0);
+  // keep track of what data the user has filled out 
+  const [wizardState, setWizardState] = useState<Partial<WizardState>>({ });
+
+  const [fullState, setFullState] = useState<WizardState | null>(null);
+  //const [fullState, setFullState] = useState<WizardState>({groupId: '', phoneticsPreference: null, collectionId: '', email: '', whereFound: ''});
+  
+  function exitWizard(){
+    const { groupId, email, whereFound } = wizardState;
+
+    // check if all required fields have been filled out
+    if (groupId && email && whereFound) {
+      // if all fields are filled out, update user state and exit the modal
+      const fullState: WizardState = {
+        groupId,
+        email,
+        whereFound,
+      };
+      steps.forEach((step) => step.commitState(fullState, userStateContext));
+      setFullState(fullState);
+      // TODO: perform any necessary cleanup or finalization
+    } else {
+      // if any required fields are missing, show an error message or prevent the user from exiting
+      alert("Please fill out all required fields.");
```

Comment:
Is this code reachable? If not, why?

Form validation is a pain in the ass to do, but if we have this branch here that surfaces an error to the user, the user should have the information they need to solve the problem. If that's too much work, we need to report an error somehow.

Eg. If we are pretty sure this code never runs, leaving a line like the below would let future developers know that we never want to hit this branch.
```
throw new Error("Expected required fields to be filled out before attempting to exit");
```

Reply:
Yes, this line can be reached and can be thrown to the user. I have added it to the exitWizardAtEnd function as well.